packages/material-ui/src/Snackbar/Snackbar.js

```diff
@@ -72,14 +64,16 @@ const SnackbarRoot = experimentalStyled(
     right: 8,
     justifyContent: 'center',
     alignItems: 'center',
-    ...(styleProps.anchorOrigin.vertical === 'top' ? top1 : bottom1),
-    ...(styleProps.anchorOrigin.horizontal === 'left' && left),
-    ...(styleProps.anchorOrigin.horizontal === 'right' && right),
+    ...(styleProps.anchorOrigin.vertical === 'top' ? { top: 8 } : { bottom: 8 }),
+    ...(styleProps.anchorOrigin.horizontal === 'left' && { justifyContent: 'flex-start' }),
+    ...(styleProps.anchorOrigin.horizontal === 'right' && { justifyContent: 'flex-end' }),
     [theme.breakpoints.up('sm')]: {
-      ...(styleProps.anchorOrigin.vertical === 'top' ? top3 : bottom3),
+      right: 'unset',
```

Comment:
Why to unset left and right? In any `anchorOrigin.horizontal` there's a left and right position.

Reply:
I was wondering the same thing. From my believe (and testing the changes using the documentation), my previous commit ([65ec44c](https://github.com/mui-org/material-ui/pull/25739/commits/65ec44cba25d1b69127ea787d4f1d50b212f2dfb)) fixed the issue. Unless I've missed something?

Reply:
I didn't see the added `left: 'auto'` and `right: 'auto'`, if those are added we don't need the `unset`.